docs/development/pigs/pig-016.rst

```diff
@@ -0,0 +1,215 @@
+obvious what kind they are.
+
+The motivation for this change is weak, but in discussions at the Erlangen
+coding sprint most Gammapy developers preferred it. The reasoning was that it's
+a bit easier to find and maintain models if they are all in one place.
+
+There was a concern that this might introduce circular dependencies between
+gammapy.modeling.models and e.g. gammapy.spectrum, gammapy.image, gammapy.cube
+or gammapy.time. But in `GH 2290`_ it was shown that this is not the case, the
+proposed change works well.
+
+There is still the open question whether models will keep this simple interface,
+or if we will add an evaluation caching layer, or whether eventually reduced
+IRFs will become models. So it is very much possible that there will be other
+changes concerning the Gammapy package structure in the future.
+
+Dissolve gammapy.background
+---------------------------
+
+We propose to dissolve gammapy.background, and move the code like this:
+
+* background_estimate.py, reflected.py, phase.py -> gammapy/spectrum
+* ring.py -> gammapy/cube/background_ring.py
+
+The motivation for this change is that currently map-related background
+estimation code is split into gammapy/background and gammapy/cube without a
+clear separation what goes where. So the choice was either to move everything to
+dissolve gammapy.background or to move all background-related code to
+gammapy.background.
+
+Looking at the package structure illustration above, it's clear that
+gammapy.background is problematic, because it is lower-level than
+gammapy.spectrum and gammapy.maps, so moving background estimation classes from
+gammapy.cube down to gammapy.background would likely lead to circular
+dependencies, because they rely on the map dataset class in gammapy.cube.
+
+The reflected region background code is not a great fit in gammapy/spectrum or
+anywhere in Gammapy except for gammapy.background, but moving it there does make
+sense if we say that all 1D spectrum code goes in gammapy.spectrum and all
+map-related analysis code goes in gammapy.cube.
+
+Dissolve gammapy.image
+----------------------
+
+The gammapy.image package was introduced early and contained a SkyImage class
+for 2D sky images, and 2D image models, and some image analysis helper functions
+such as code to make a radial profile from an image. Much later gammapy.maps was
+introduced and we managed to support 2D and 3D maps with one container instead
+of separate gammapy.image.SkyImage and gammapy.cube.SkyCube.
+
+Now, if we move gammapy.image.models to gammap.modeling as proposed above, and
+that SkyImage has been removed in 2018, there is very little functionality left
+in gammapy.image, and it's not clear what it's scope is.
+
+So we propose to clean this up, roughly like this:
+
+- asmooth.py and utils.py -> move to gammapy.cube
+- measure.py and profile.py -> replace with docs example or move to maps or cube
+- plotting.py -> replace with docs example or move to maps
```

Comment:
Or add a plot sub package in the future?

Reply:
Make a decision here which functionality to keep and move and which to replace by docs examples? My proposal would be:

- Keep `ImageProfile` and `ImageProfileEstimator`, rename it to `MapProfile` and `MapProfileEstimator` and move it to `gammapy.maps` or `gammapy.cube`. This opens the way to extend it to arbitrary extra axes in future and implement e.g. likelihood fitting based algorithms.
- Remove `illustrate_colomap` and `grayify_colormap` from `image.plotting`, keep the other colormaps and the `MapPanelPlotter` but move it to `gammapy.maps.visualization`
- Move `ASmooth` to `gammapy.detect`, move `image.utils` to `maps.utils`
- Move `image.measure` to `maps.measure` for now, dissolve it maybe later.

Reply:
@adonath - I've updated to this:
```
- utils.py -> one helper function, move to gammapy/maps/utils.py
- asmooth.py -> move to gammapy.detect
- measure.py and profile.py -> replace with docs example or move to maps or cube
- plotting.py -> move to gammapy/maps/plotting.py (remove illustrate_colormap and grayify_colormap)
```

For measure.py and profile.py we disagree at the moment. I think their usefulness and quality isn't good enough to keep, and we shouldn't spend time on it. Also it's not clear to me why you propose to put measure.py in gammapy.maps and profile.py in gammapy.cube, to me the code level seems very similar and naively I wouldn't expect one in one sub-package and the other somewhere else.

Would it be OK to keep this question in the PIG for now? It just means we don't have to agree on the solution now, and don't make these details part of the review by others, but it's something we will decide later on PR review.

Reply:
@adonath - Changed to this following your suggestion:
```
- measure.py and profile.py -> move to gammapy.maps (and clean up)
```